text_adventure.py

```diff
@@ -0,0 +1,190 @@
+import random
+
+class GameObject:
+    verb_dict = {
+        "examine": "Examine a creature in the room.",
+        "hit": "Hit a creature in the room.",
+        "help": "Get help about the available commands.",
+        "heal": "Heal for an amount of damage.",
+        "kill": "Immediately kill the target.",
+        "status": "Get your own status."
+    }
+
+    @staticmethod
+    def get_input():
+        command = input(": ").split()
+        verb_word = command[0]
+        if verb_word in GameObject.verb_dict:
+            verb = getattr(GameObject, verb_word)
+        else:
+            print(f"Unknown verb {verb_word}")
+            print("For a list of commands type \"help\"")
+            return
+    
+        if len(command) >= 2:
+            noun_word = command[1]
+            print(verb(noun_word))
+        else:
+            print(verb("nothing"))
+
+    @staticmethod
+    def hit(noun):
+        for creature in character.current_room.creatures:
```

Comment:
Think about what the hit method is doing, it's hitting some single thing right? so it should only need to do 1 unit of work. With this approach, this function takes time proportional to the number of units in the room, can you think of a better approach?